docs/design/Non_Admin_Controller_design.md

```diff
@@ -0,0 +1,133 @@
+# Non-Admin Backup/Restore Design
+
+## Background
+OADP (Openshift API for Data Protection) Operator currently requires cluster admin access for performing Backup and Restore operations of applications deployed on the OpenShift platform. This design intends to enable the ability to perform Backup and Restore operations of their own application namespace for namespace owners aka non-admin users.
+
+## Goals
+- Enable non-admin backup operation
+- Enable non-admin restore operation
+
+## Non-Goals
+- Performance improvements of Backup and Restore Operations
+- Parallel Backup and Restore Operations
+
+## Use-Cases
+
+### Backup Operation
+- As a non-admin user/namespace owner with administrative priviledges for a particular namespace, the user should be able to:
+    - Create a Backup/Schedule of the namespace
+    - Update the Backup/Schedule spec of the namespace
+    - View the status of the Backup/Schedule created for the particular namespace
+    - Delete the Backup/Schedule of the namespace
+
+### Restore Operation
+- As a non-admin user/namespace owner with administrative priviledges for a particular namespace, the user should be able to:
+    - Create a Restore of the namespace
+    - Update the Restore spec of the namespace
+    - View the status of the Restore created for the particular namespace
+    - Delete the Restore of the namespace
+
+
+## Installation
+
+- The Non-Admin Controller (NAC) will be installed via OADP Operator. 
+- The Data Protection Application (DPA) CR will consist of a root level spec flag called `enableNonAdmin`
+- If the `enableNonAdmin` flag is set to `true`, the OADP Operator will install the NAC in OADP Operator's install namespace, the default value for `enableNonAdmin` will be `false` 
```

Comment:
We need some validation to ensure that only one DPA in the cluster sets this to `true`. Otherwise, if there are 2 OADP instances, both enabling non-admin, we will have race conditions around which velero instance handles each NonAdminBackup.

Reply:
> if there are 2 OADP instances, both enabling non-admin

Both instances in a supported scenario would be in separate oadp namespaces, I don't see a possible overlap here.

Reply:
```
by default it watches all Namespaces
```
nvm. agree with sseago.

Reply:
Right. scenario is this:
- OADP installed in openshift-adp
- OADP installed in openshift-adp2
- user creates NonAdminBackup in mysql-persistent

If both OADPs have non-admin enabled, it's a race condition as to which one grabs and labels/annotates it first (and  creates Velero CR), and possibly in some cases you end up with both attempting to modify it at the same time. If only one of them has non-admin enabled, then that OADP will manage it, and the other OADP will stay out of the way.

Reply:
So I think non admin use case will prop up only in env where multiple users share a cluster. Therefore I think we need to ensure that multiple non admin controllers per velero is supported, so we need a way to limit the scope of each nac so it doesn't overlap.